siliconcompiler/core.py

```diff
@@ -4211,6 +4211,18 @@ def _local_process(self, flow, status, steplist, indexlist):
         for (step, index) in flowgraph_nodes:
             # Setting up tool is optional
             self._setup_node(step, index)
+            # Env vars are necessary as test_multiple_tools.py requires it for its version check
+            self._set_env_vars(step, index)
+            tool, task = self._get_tool_task(step, index, flow)
+            # Icarus compiles its executable during the run so we can't check its version
+            if (tool, task) != ('execute', 'exec_input'):
+                run_func = getattr(self._get_task_module(step, index, flow=flow), 'run', None)
+                try:
+                    self._check_tool_version(step, index, run_func)
```

Comment:
It might be nice to only get errors from this function when calling in this context? Otherwise it's a lot of printing.

Also this could be done in parallel to speed things up a bit

Reply:
Sure for now it is definitely printing the version checks here **and** when running the node so I could not log that here.
I don't think parallelizing here seems worth the additional complexity.

Reply:
I think we should check this on some larger graphs on the VMs.

Reply:
Created a task for this: SC-638